subprojects/ivy/src/integTest/groovy/org/gradle/api/publish/ivy/IvyPublishJavaIntegTest.groovy

```diff
@@ -699,6 +699,51 @@ class IvyPublishJavaIntegTest extends AbstractIvyPublishIntegTest {
         }
     }
 
+    def "can publish java component with capabilities"() {
+        given:
+        createBuildScripts("""
+            dependencies {
+                capabilities {
+                    capability('org:cap') {
+                        providedBy 'org:foo'
+                    }
+                    capability('org:cap2') {
+                        providedBy 'org:foo'
+                        providedBy 'org:bar'
+                        prefer 'org:foo'
+                    }
+                    capability('org:cap3') {
+                        providedBy 'org:foo'
+                        providedBy 'org:bar'
+                        providedBy 'org:baz'
+                        prefer 'org:bar' because 'sets a preference'
```

Comment:
I find it interesting that what your test publishes would break all users. Should we have some internal validation?

Reply:
I'm not sure what you mean: there are 3 different capabilities published by the same module here.

Reply:
Yes but two of these capabilities are provided by `org:foo` and `org:bar`, and one says `prefer org:bar` while the other says `prefer org:foo` which means that any project having both `org:foo` and `org:bar` as dependencies will fail unless the metadata is fixed. Makes sense now?

Reply:
Right, but it's not necessarily a mistake. All it says is that you can't have both at the same time, which may be exactly what you want to say.